scene/3d/navigation.cpp

```diff
@@ -404,14 +379,30 @@ Vector<Vector3> Navigation::get_simple_path(const Vector3 &p_start, const Vector
 		Polygon *p = least_cost_poly->get();
 		//open the neighbours for search
 
+		if (p == end_poly) {
+			//oh my reached end! stop algorithm
+			found_route = true;
+			break;
+		}
+
 		for (int i = 0; i < p->edges.size(); i++) {
 
 			Polygon::Edge &e = p->edges.write[i];
 
 			if (!e.C)
 				continue;
 
+#ifdef USE_ENTRY_POINT
```

Comment:
I wonder what's the point of this define since it can't be toggled conditionally. Sounds like we're maintaining dead code in the `#else` branches.

Reply:
It was added in #22047, so I decided to not touch it yet.